Added deposit incentive feature

JonathanGinsberg opened the pull request:
**Changes Made in this PR**
I created an interest feature that for every 5 deposits of an amount greater than or equal to $20 the customer makes, the customer will receive an interest of 1.5%. This will incentivize users to develop a habit in depositing more frequently into their account and will attract more customers to testudo bank.

`src/main/java/net/codejava/MvcController.java`

- Modified `applyInterest()` so that each time when a customer deposits 5 times, interest rate will be applied to their current balance and a new row will be added to the transaction table indicating this change. After the previous changes are made, `NumDepositsForinterest` will be reset to zero.
---
`src/test/java/net/testudobank/tests/MvcControllerIntegTest.java`

- Implemented a test to verify that interest is not applied when a customer has overdraft and a balance that is not greater than 0.
- Implemented a test to verify that interest is not applied for 5 deposits under $20.
- Implemented a test to verify that interest is applied to customer's balance with 5 deposits of $20 and no overdraft with 0 balance.
---

s1ddh4rthc commented:
Hey Jonathan, nice start to Assignment 2. I'd say to keep in mind that your commit messages should be more meaningful, so make sure that your commit messages focus on the why and not the what. Your logic looks good, just make sure to finish the integ tests for part 2. Also, your PR message looks good, and you can make improvements if you search up how to better format your PR message on GitHub.

JonathanGinsberg commented:
**Changes Made in this PR**
Added two additional test cases to verify edge cases where user input is slightly above or below the required amount.

`src/test/java/net/testudobank/tests/MvcControllerIntegTest.java`

- Implemented a test to verify that the feature doesn't apply interest to the user's balance when the user does 5 deposits of $19.98.
- Implemented a test to verify that the feature does apply interest to the user's balance when the user does 5 deposits of $20.01.
---
